src/gc.h

```diff
@@ -100,13 +103,13 @@ class GarbageCollector
         serverAssert(m_setepochOutstanding.find(epoch) != m_setepochOutstanding.end());
         serverAssert(sp->FWillFreeChildDebug() == false);
 
-        auto itr = std::find(m_vecepochs.begin(), m_vecepochs.end(), m_epochNext+1);
-        if (itr == m_vecepochs.end())
+        auto itr = std::find(m_listepochs.begin(), m_listepochs.end(), m_epochNext+1);
```

Comment:
Would it be possible to use an unordered set? You could still iterate in O(N) and insert/delete in O(1), but this find operation would be O(1) instead of O(N)?

Reply:
We rely on sorted order though

Reply:
What about an ordered set? insertion/deletion would be logarithmic instead of constant, but so would this find.

Reply:
Deletion is O(log n) though, this would regress us to O(n log n) performance for O(n) that we are currently so its not a net benefit.